addons/payment_stripe/models/payment.py

```diff
@@ -49,14 +49,46 @@ def stripe_form_generate_values(self, tx_values):
             'payment_intent_data[description]': tx_values['reference'],
             'customer_email': tx_values.get('partner_email') or tx_values.get('billing_partner_email'),
         }
-        if tx_values.get('billing_partner_country').code and tx_values.get('billing_partner_country').code.lower() == 'nl' and \
-           tx_values.get('currency').name and tx_values.get('currency').name.lower() == 'eur':
-            # enable iDEAL for NL-based customers (€ payments only)
-            stripe_session_data['payment_method_types[1]'] = 'ideal'
+
+        self._add_available_payment_method_types(stripe_session_data, tx_values)
+
         tx_values['session_id'] = self._create_stripe_session(stripe_session_data)
 
         return tx_values
 
+    @api.model
+    def _add_available_payment_method_types(self, stripe_session_data, tx_values):
+        """
+        Add payment methods available for the given transaction
+
+        :param stripe_session_data: dictionary to add the payment method types to
+        :param tx_values: values of the transaction to consider the payment method types for
+        """
+        PMT = namedtuple('PaymentMethodType', ['name', 'countries', 'currencies', 'recurrence'])
+        all_payment_method_types = [
+            PMT('card', [], [], 'recurring'),
+            PMT('ideal', ['nl'], ['eur'], 'punctual'),
+            PMT('bancontact', ['be'], ['eur'], 'punctual'),
+            PMT('eps', ['at'], ['eur'], 'punctual'),
+            PMT('giropay', ['de'], ['eur'], 'punctual'),
+            PMT('p24', ['pl'], ['eur', 'pln'], 'punctual'),
+        ]
+
+        country_checked = filter(lambda pmt: not pmt.countries or
```

Comment:
can you improve the readability a bit more please? e.g. 
```py
country_code = tx_values.get('billing_partner_country', self.env['res.country']).code or `no_country`
country_filtered_pm = filter(lambda pm: not pm.countries or country_code.lower() in pm.countries, all_payment_method_types)
```

Reply:
@dbo-odoo done